src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container21.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+require_once __DIR__.'/../includes/classes.php';
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+$container = new ContainerBuilder();
+
+$container->setParameter('factory.service', 'foo_factory');
+$container->setParameter('factory.class', 'FooFactory');
+$container->setParameter('factory.method', 'createFoo');
+
+$container
+    ->register('foo1', 'Foo')
+    ->setFactory(array('%factory.class%', '%factory.method%'))
+    ->setPublic(false)
+;
+$container
+    ->register('foo2', 'Foo')
+    ->setFactory(array(new Definition('%factory.class%'), '%factory.method%'))
+    ->setPublic(false)
+;
+$container
+    ->register('foo3', 'Foo')
+    ->setFactory(array(new Reference('%factory.service%'), '%factory.method%'))
```

Comment:
this is invalid. References cannot use parameters anywhere in Symfony (you should use aliases if you want to make references configurable rather than parameters)